Quiz creation: Snackbar improvements

AlexVelezLl opened the pull request:
## Summary
Replace snackbar support in route.query with direct calls to the `createSnackbar` action. And ensures that these are not deleted when entering any of the `skipLoading` routes

## References
Closes #12038.

## Reviewer guidance
<!--
 * how can a reviewer test these changes?
 * are there any risky areas that deserve extra testing
-->

Play around with the Quiz Editor saving sections, replacing questions, removing sections, go back from the Side Panel, and snackbar should behave consistently.


## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests


## PR process

- [x] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

AlexVelezLl committed:
Snackbar improvements

AlexVelezLl committed:
Remove snackbar query support

nucleogenesis reviewed (approved):
I tested the snackbars and they all work perfectly. This is good to merge IMO, thank you @AlexVelezLl 

I did note one bug unrelated to your changes in #12037 (I confirmed it is unrelated by replicating it on develop).

marcellamaki commented:
@radinamatic @pcenov if you could do a QA double-check here, then we will be good to go! thank you :) there should be some notes of what to test in the issue that Alex linked in the description

pcenov commented:
Hi @AlexVelezLl and @nucleogenesis - I am not seeing any snackbars in the following scenarios:

- user deletes questions from the list
- user rearranges questions in the list
- user rearranges the section order in the 'Section settings' modal and closes the modal without clicking 'Apply settings'

https://github.com/learningequality/kolibri/assets/79847249/d71b3844-09cf-479b-b914-8ad7ad2f4d97

Not sure if this in scope for this PR - if not, I can file a separate issue.

rtibbles reviewed (dismissed):
Seems like @AlexVelezLl and @nucleogenesis do indeed want the three scenarios @pcenov has described to be covered here, so blocking merge until that has happened.

AlexVelezLl committed:
Add missing snackbars notifications

AlexVelezLl committed:
add sections order changed check

AlexVelezLl commented:
Hi @nucleogenesis @pcenov I've added the missing snackbar notifications: 

https://github.com/learningequality/kolibri/assets/51239030/f84b2d67-ec6f-4780-9a32-69de4c223aaf

@nucleogenesis I saw that the changes to the section order were made immediately after the drag and drop, and it didn't wait until the user clicked on the "apply changes" button, so I added a new ref for the section order within the component and added a `sectionOrderChanged` check before saving the changes [here](https://github.com/learningequality/kolibri/pull/12080/commits/48f85a4d7333f846ede92cfb816e95c2c85e68b0). Let me know if this is correct.

pcenov commented:
Thanks @AlexVelezLl - I confirm that the snackabar notifications are displayed now in the first 2 cases, and that the section order change is applied only after the user has clicked the 'Apply settings' button.

radinamatic reviewed (approved):
Manual QA passes, good to go! 💯 👏🏽 :shipit:
